core/src/main/java/org/apache/carbondata/core/datastore/page/SafeFixLengthColumnPage.java

```diff
@@ -431,6 +439,15 @@ private void ensureArraySize(int requestSize, DataType dataType) {
         System.arraycopy(doubleData, 0, newArray, 0, arrayElementCount);
         doubleData = newArray;
       }
+    } else if (dataType == DataTypes.BYTE_ARRAY) {
```

Comment:
increasing by 16 is too low, it can be doubled like array list case

Reply:
Fixed